compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -895,9 +895,16 @@ private boolean isAssignable(BType source, BType target, Set<TypePair> unresolve
             return isFiniteTypeAssignable((BFiniteType) source, target, unresolvedTypes);
         }
 
-        if ((targetTag == TypeTags.UNION || sourceTag == TypeTags.UNION) &&
-                isAssignableToUnionType(source, target, unresolvedTypes)) {
-            return true;
+        if ((targetTag == TypeTags.UNION || sourceTag == TypeTags.UNION)) {
+            boolean isAssignableToUnion = isAssignableToUnionType(source, target, unresolvedTypes);
+            if (isAssignableToUnion) {
+                return true;
+            }
+
+            TypePair pair = new TypePair(source, target);
+            if (unresolvedTypes.contains(pair)) {
+                unresolvedTypes.remove(pair);
+            }
```

Comment:
We can remove `contains` check

Reply:
Yes. I will remove.

Reply:
https://github.com/ballerina-platform/ballerina-lang/blob/9355b815333b7b8b874bb7245d0a816259f6df45/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java#L3441

In here only cyclic unions are added to unresolvedTypes.
So I think we can't remove contains check.

Reply:
The `remove ` method check element is present in the Set and then remove it. (otherwise, it returns False) So do we need a `contains` check?

Reply:
Yes we don't need that. Removed in 4c82e4f1dfb0dba993c5a76e28b7ee7497fdc204